client/state/signup/steps/domains/reducer.js

```diff
@@ -0,0 +1,30 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+
+import { SIGNUP_STEPS_DOMAIN_SEARCH_PREFILL_SET } from 'state/action-types';
+
+import { createReducer } from 'state/utils';
+import { domainStepSchema } from './schema';
+
+export default createReducer(
+	{},
+	{
+		[ SIGNUP_STEPS_DOMAIN_SEARCH_PREFILL_SET ]: ( state, action ) => {
+			const { overwrite, prefill } = action;
+
+			// Only set prefill if the value hasn't already been set.
+			if ( typeof state.prefill === 'string' && state.prefill.length > 0 && ! overwrite ) {
```

Comment:
Probably overkill to test if it's a string?
Maybe we can default to it in the action?

```es6
export function setDomainSearchPrefill( prefill = '', overwrite = false ) { ... }
```